### PR TITLE
Updates and more structure

### PR DESCRIPTION
**Pull request by h0st1le**
First things first: Thanks for open sourcing DIWire Bender. That being said (and I hope you guys do not take offence to this) but in general I feel that the documentation could really use some organization. Most of the information is scattered through a variety of locations, Pensa's Blog, Google Code repo, and multiple BOM files. In addition the repo on github doesnt seem to be in sync with the official repo on google code. To remedy this I set out restructuring your git repo in an attempt to bring all the information into a single location easily accessible, viewable, and forkable.

TLDR; So i did some serious reorganization to your repo, added in an assload of readme info, updated BOM and added stls of all printable parts.

1.) Pulled latest code from the google code repository; merged changes into existing repository.
2.) Added all files from the latest BOM download archive
3.) Converted all STEP files to STL for easy 3d printing
4.) Added in additional parts from older BOM that were missing from the new BOM archive (was this an accident)
5.) Renamed all 3d-printed parts to fit standard naming convention
6.) Added in gplv3 and cc-by-sa licenses
7.) Migrated readme document to markdown for display on github and added in several new sections
8.) Change BOM format to use a similar format to that of kitBOM
9.) Migrated BOM from excel document to markdown (again for display on github)

### CODE REVIEW

**Comment by diwire** [comment]
Justin,
Thanks so much for all the hard work! I have to spend some time to go
through it. I really appreciate all the help!
Stay in touch,
Marco

On Wed, Feb 20, 2013 at 8:19 PM, Justin Penka notifications@github.comwrote:

> First things first: Thanks for open sourcing DIWire Bender. That being
> said (and I hope you guys do not take offence to this) but in general I
> feel that the documentation could really use some organization. Most of the
> information is scattered through a variety of locations, Pensa's Blog,
> Google Code repo, and multiple BOM files. In addition the repo on github
> doesnt seem to be in sync with the official repo on google code. To remedy
> this I set out restructuring your git repo in an attempt to bring all the
> information into a single location easily accessible, viewable, and
> forkable.
> 
> Here is what I did:
> 
> 1.) Pulled latest code from the google code repository; merged changes
> into existing repository.
> 2.) Added all files from the latest BOM download archive
> 3.) Converted all STEP files to STL for easy 3d printing
> 4.) Added in additional parts from older BOM that were missing from the
> new BOM archive (was this an accident)
> 5.) Renamed all 3d-printed parts to fit standard naming convention
> 6.) Added in gplv3 and cc-by-sa licenses
> 7.) Migrated readme document to markdown for display on github and added
> in several new sections
> 8.) Change BOM format to use a similar format to that of kitBOM
> 9.) Migrated BOM from excel document to markdown (again for display on
> github)
> 
> TLDR; So i did some serious reorganization to your repo, added in an
> 
> ## assload of readme info, updated BOM and added stls of all printable parts.
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/h0st1le/DIWire-Bender master
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/diwire/DIWire-Bender/pull/1
> Commit Summary
> - Changes in header
> - Changes in Header
> - Fixed Typo in header
> - New Header
> - Header fix - what this file does
> - Header adjustment - what this file does.
> - header typo
> - header typo
> - header typo
> - Make Your Own DIWire Video at https://vimeo.com/44274793 for
>   construction details
> - added latest BOM and 3d printed parts
> - converted all STEP files to STL for easy 3d printing
> - latest printed parts download was missing Angle_Marker and
>   Top_Plate_Hole_Layout, pulled from previous version and added
> - renamed Top_Plate_Hole_layout and Angle_Marker to fit existing
>   naming scheme
> - added in proper licenses, migrated README to markdown and added
>   several chunks of new information
> - added linebreaks to license.txt and stripped all trailing
>   whitespaces from all other files
> - small fixes to readme
> - added gitignore, feed it with vim file types
> - changed format of the BOM to match that of kitBOM also moved the BOM
>   into markdown. Lastly added in all items that appear to be removed from
>   latest BOM download into Additional Materials supplimentary list
> - added section 'what can it do' some fixes and cleanup
> - typo fix
> - another typo fix
> - just found the instructable, added it as resource to readme
> 
> File Changes
> - _A_ .gitignorehttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-0(6)
> - _M_ 2D-Processing.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-1(73)
> - _A_ 3D-Printed-Parts/STEP/Pensa000_Top_Plate_Hole_Layout.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-2(2232)
> - _A_ 3D-Printed-Parts/STEP/Pensa001_Motor_Mount.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-3(2580)
> - _A_ 3D-Printed-Parts/STEP/Pensa002_Solenoid_Mount.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-4(2808)
> - _A_ 3D-Printed-Parts/STEP/Pensa003_Bender_Head.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-5(1379)
> - _A_ 3D-Printed-Parts/STEP/Pensa004_Straightener_Wheel.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-6(9647)
> - _A_ 3D-Printed-Parts/STEP/Pensa005_Motor_Bracket.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-7(1118)
> - _A_ 3D-Printed-Parts/STEP/Pensa006_Angle_Marker.STEPhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-8(9450)
> - _A_ 3D-Printed-Parts/STL/Pensa000_Top_Plate_Hole_Layout.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-9(0)
> - _A_ 3D-Printed-Parts/STL/Pensa001_Motor_Mount.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-10(0)
> - _A_ 3D-Printed-Parts/STL/Pensa002_Solenoid_Mount.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-11(0)
> - _A_ 3D-Printed-Parts/STL/Pensa003_Bender_Head.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-12(0)
> - _A_ 3D-Printed-Parts/STL/Pensa004_Straightener_Wheel.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-13(0)
> - _A_ 3D-Printed-Parts/STL/Pensa005_Motor_Bracket.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-14(0)
> - _A_ 3D-Printed-Parts/STL/Pensa006_Angle_Marker.stlhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-15(0)
> - _M_ 3D-Processing.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-16(49)
> - _M_ 3D-ReadMe.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-17(8)
> - _M_ Arduino.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-18(70)
> - _A_ BOM.mdhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-19(39)
> - _A_ DIWire Bender BOM.xlsxhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-20(0)
> - _A_ LICENSE.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-21(17)
> - _A_ Misc/cc-by-sa-3.0-us.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-22(359)
> - _A_ Misc/gpl-3.0.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-23(674)
> - _A_ README.mdhttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-24(65)
> - _D_ ReadMe.txthttps://github.com/diwire/DIWire-Bender/pull/1/files#diff-25(12)
> 
> Patch Links:
> - https://github.com/diwire/DIWire-Bender/pull/1.patch
> - https://github.com/diwire/DIWire-Bender/pull/1.diff

## 

marco perry . principal, pensa
20 jay st., suite 800, brooklyn, ny 11201
p 718-855-5354 .  blog.pensanyc.com http://www.pensanyc.com/  .
www.pensanyc.com . @thinkpensa

**Comment by MFlego** [comment]
Hello,  I want to contribute to this project, Its been quite a while since anyone has worked on it but I think its time to resurrect it.  I will be working on new 3d files mostly and trying to use the source code.

**Comment by h0st1le** [comment]
Pensa actually ended up doing quite a bit of work on this project,
including a kickstarter (
https://www.kickstarter.com/projects/1638882643/diwire-the-first-desktop-wire-bender).
They appear to be selling the final version on their homepage (
http://www.pensalabs.com/#home) but it doesn't appear they open sourced any
of the improvements or changes. I originally submitted a pull request to
this project in attempt to centralize all the information I could find on
it from various sources (instructables, Pensa's Blog, the Google Code repo
and the multiple BOM files). I never got a chance to go back and do any
more work with the project but you may want to check the pull request I
submitted for some additional info not contained in the original repo. I
would really like to see this project pick up again, excited to see what
progress you make with it. - Cheers

**Comment by diwire** [comment]
That would be great to see it improved! Today there are simpler motor
motion boards for arduinos or Pi. That would be a better way to do it.

On Friday, April 24, 2015, Matthew Flego notifications@github.com wrote:

> Hello, I want to contribute to this project, Its been quite a while since
> anyone has worked on it but I think its time to resurrect it. I will be
> working on new 3d files mostly and trying to use the source code.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/diwire/DIWire-Bender/pull/1#issuecomment-96124756.

## 

marco perry . principal, pensa & pensa labs
20 jay st., suite 800, brooklyn, ny 11201
p 718-855-5354 .  @thinkpensa
www.fb.com/pensa  . www.fb.com/pensalabs
blog.pensanyc.com http://www.pensanyc.com/  . www.pensanyc.com .
www.pensalabs.com  http://www.pensalabs.com
